test/staging/Intl402/Temporal/old/date-time-format.js

```diff
@@ -616,42 +639,56 @@ assert.deepEqual(at.formatToParts(end), [
 
 //should work for Time 
-assert.sameValue(us.formatRange(Temporal.PlainTime.from(t1), Temporal.PlainTime.from(t2)), "2:23:30 PM \u2013 3:44:56 PM");
-assert.sameValue(at.formatRange(Temporal.PlainTime.from(t1), Temporal.PlainTime.from(t2)), "14:23:30 \u2013 15:44:56");
+assert.sameValue(
+  us.formatRange(Temporal.PlainTime.from(t1), Temporal.PlainTime.from(t2)),
+  `2:23:30${amPmSpace}PM${dateRangeSeparator}3:44:56${amPmSpace}PM`
+);
+assert.sameValue(at.formatRange(Temporal.PlainTime.from(t1), Temporal.PlainTime.from(t2)), `14:23:30${dateRangeSeparator}15:44:56`);
 
 //should work for Date 
-assert.sameValue(us.formatRange(Temporal.PlainDate.from(t1), Temporal.PlainDate.from(t2)), "11/18/1976 \u2013 2/20/2020");
-assert.sameValue(at.formatRange(Temporal.PlainDate.from(t1), Temporal.PlainDate.from(t2)), "18.11.1976 \u2013 20.02.2020");
+assert.sameValue(us.formatRange(Temporal.PlainDate.from(t1), Temporal.PlainDate.from(t2)), `11/18/1976${dateRangeSeparator}2/20/2020`);
+assert.sameValue(at.formatRange(Temporal.PlainDate.from(t1), Temporal.PlainDate.from(t2)), `18${germanDateSeparator}11${germanDateSeparator}1976${dateRangeSeparator}20${germanDateSeparator}02${germanDateSeparator}2020`);
 
 //should work for YearMonth 
 var date1 = Temporal.PlainDate.from(t1);
 var date2 = Temporal.PlainDate.from(t2);
-assert.sameValue(us.formatRange(date1.withCalendar(usCalendar).toPlainYearMonth(), date2.withCalendar(usCalendar).toPlainYearMonth()), "11/1976 \u2013 2/2020");
-assert.sameValue(at.formatRange(date1.withCalendar(atCalendar).toPlainYearMonth(), date2.withCalendar(atCalendar).toPlainYearMonth()), "11.1976 \u2013 02.2020");
+assert.sameValue(us.formatRange(date1.withCalendar(usCalendar).toPlainYearMonth(), date2.withCalendar(usCalendar).toPlainYearMonth()), `11/1976${dateRangeSeparator}2/2020`);
+assert.sameValue(at.formatRange(date1.withCalendar(atCalendar).toPlainYearMonth(), date2.withCalendar(atCalendar).toPlainYearMonth()), `11${germanMonthYearSeparator}1976${dateRangeSeparator}02${germanMonthYearSeparator}2020`);
 
 //should work for MonthDay 
 var date1 = Temporal.PlainDate.from(t1);
 var date2 = Temporal.PlainDate.from(t2);
-assert.sameValue(us.formatRange(date2.withCalendar(usCalendar).toPlainMonthDay(), date1.withCalendar(usCalendar).toPlainMonthDay()), "2/20 \u2013 11/18");
-assert.sameValue(at.formatRange(date2.withCalendar(atCalendar).toPlainMonthDay(), date1.withCalendar(atCalendar).toPlainMonthDay()), "20.02. \u2013 18.11.");
+assert.sameValue(
+  us.formatRange(date2.withCalendar(usCalendar).toPlainMonthDay(), date1.withCalendar(usCalendar).toPlainMonthDay()),
+  `2/20${dateRangeSeparator}11/18`
+);
+// In Node 19.2.0, apparently the german Month/Day format uses regular spaces, but all other formats and locales seem to use thin spaces ('\u{2009}');
```

Comment:
There seems to be a lot of variability with de-AT; perhaps it would make sense to pick a different non-English locale if these comparisons truly are important?

Reply:
After quite a bit of digging, the problem looks like a regression in CLDR 42's German date formats. Frustratingly, the Node 19 build used by GitHub Actions picks up ICU 72 (which includes CLDR 42 with the problems) but `brew install node` on my local machine currently includes ICU 71/CLDR 41. This explains why I only saw it in CI and why this problem has been so frustrating to track down!

I filed a bug in CLDR's JIRA to fix the formats: https://unicode-org.atlassian.net/browse/CLDR-16243

Now that I understand the actual root cause, I can remove a lot of the German-specific code because only month/year formats are affected, not all date formats. This should clean up this PR quite a bit. I'll do that tomorrow and will ping you to review again once that's done.

Thanks for pushing me to dig into this!

Reply:
I ended up leaving German as the locale because AFAICT would be a lot of work to change it throughout all these tests, and it's relatively straightforward to write tests that ignore the German issues. I added TODO comments that encourage future contributors to consider tests that have less dependence on CLDR formats.